### DISCUSSION TITLE
sensor based on PHY6252 / V1.5

### DISCUSSION THREAD

**Discussion by FrankyGT** [pvvx/THB2#82] [Q&A]
Hi, I bought some temperature sensors last week, which , from the display image, look like a TH05 V1.4. But inside is a PHY6252 chip as well as a VKL060 LCD driver chip.  And there is only one marking under the LCD panel  : 

QC0000_L11_PHY6252_TEMP_HUM_V1.5

So it looks like V1.5 has been rolled out...

**Comment by pvvx** [comment]
I don't have any thermometers with [PHY6252](https://github.com/pvvx/PHY62x2/blob/master/Doc/PHY6252_BLE_SoC_Datasheet_v1.3.pdf).

**Comment by gmag11** [comment]
I attach a few pictures of this model:

Design is very simple. Just a few discrete components. No battery included :(
![image](https://github.com/user-attachments/assets/5fb832aa-a524-4e38-b948-002c8f5e8dd7)

SOC is PHY6252
![image](https://github.com/user-attachments/assets/8aabdd6e-a6be-4936-b836-befd1a7982e8)
Datasheet is [here](https://docs.ai-thinker.com/_media/ble/phy6252_ble_soc_datasheet_v1.3_20221017.pdf)

LCD display driver is VKL060. I haven't found any info about it
![image](https://github.com/user-attachments/assets/7e3d6001-5f0b-4b04-a434-f465d463d9ce)

This is the only reference in the board
![image](https://github.com/user-attachments/assets/97fd59a7-05d4-402d-b037-38b7004fd1cc)

Sensor seems to be a clone of [AHT20](https://files.seeedstudio.com/wiki/Grove-AHT20_I2C_Industrial_Grade_Temperature_and_Humidity_Sensor/AHT20-datasheet-2020-4-16.pdf)
![image](https://github.com/user-attachments/assets/7c1e8ac2-a518-4df7-94f1-37b3170ead72)

These seem to be GND, TX, RX and Vcc test points for flashing.
![image](https://github.com/user-attachments/assets/77d32ca9-2841-49a2-a225-b3a1f3845fac)

**Comment by pvvx** [comment]
The PHY6252 is a stripped-down, cheaper version of the PHY6222.
It consumes a lot of battery power - there is no onboard DC-DC to power the CPU core.
The PHY6252's sleep current consumption is 13 µA. Plus the LCD driver chip and sensor. That's over 20 µA in total (sleep mode).

Combined with BLE, sensor and screen processing, the consumption will be too high for the CR2032.

I wouldn't use a thermometer like that. These chips (PHY6252) are used in light bulbs and other devices powered by the mains.

**Reply by FrankyGT** [reply]
Wel according to the specs the power usage increase is indeed 150% of the PHY6222. But only during sending/receiving data. This does translate though to a lower transceiver sensitivity on the PHY6252. I would guess this in effect would translate to better range/stability. But I havent got any PHY222 sensors so I cant verify this...

**Reply by pvvx** [reply]
The data in the PDF is incorrect and depends on many conditions.
Real measurements show a significant increase in consumption.
I have had PB-03F modules with PHY6252 chips for a long time.
Depending on the version of the PHY documentation, different parameters are specified. It's a complete mess.

**Reply by pvvx** [reply]
In sleep mode, DC-DC is disabled.
It is possible that in packages without DC-DC pins, the PHY installs chips crystal with higher leakage.
Leaks for RAM support and RC generator operation in normal PHY6222 crystals are at the level of 2..5 µA. Depends on the quality of the crystal.

Additionally, without DC-DC in SoC activity mode, consumption will be 2..3 times higher. (3.3V -> DC-DC -> 1.2V)

Accordingly, when CR2032 is discharged (increasing internal resistance), we get a voltage drop of 2.5 times greater. Chemically, the CR2032 element produces 3V at +25C. Discharge of the CR2032 battery = increase in internal resistance. And the current of PHY6252 during active operation is 2+ times higher. As a result, we get a limitation of battery use to different levels of discharge and large voltage surges when switching from work to sleep.

With a pulse load of 8..10 mA, the average CR2032 battery gives only 100 mAh.
That is, with an internal voltage of 3V and a current of 5 mA, the CR2032 battery will produce 2.2V. The internal resistance of the battery will be 160 Ohm (voltage drop of 0.8V).
The active mode of the SoC for BLE (for PHY chips) operation cannot be less than 5 ms.
When using a CRxxx battery with a large mAh, the dependencies are completely different and non-linear. Current BLE SoC technologies use CR2032 at the limit of maximum pulse currents. For a CR2450 battery, the pulse discharge dependence will be completely different, significantly different from the ratio of the capacity in mAh given by the manufacturer.
Then you can calculate everything yourself.

You were sold a cheap dummy thermometer. There is no point in wasting time on such things from the Chinese market.

**Reply by gabbla** [reply]
Just for fun (and because I didn't find this thread sooner) I ported the fw on such device. It took a while to map the LCD but I have a working porting. I see around 18 uA drained from a CR2032 battery (without any link). @pvvx should I take the time to clean stuff out and open a PR?

**Reply by pvvx** [reply]
The ST17H66 chip is almost an analogue of Phy6252.
Current in sleep mode with active RAM 11..12 uA at 3.3 V. (Deep-sleep < 2 uA)
Since there is no built-in DC-DC, RF transmission has a larger current of 2..3 times than that of PHY6222.
At RF TX +0 dBm the pulse current is 13 mA. An example of what voltage will be on a partially discharged CR2032, but still quite functional:
![image](https://github.com/user-attachments/assets/16590c4b-df46-4e5f-b8ac-6de911e7ceef)

On the new CR2032:
![image](https://github.com/user-attachments/assets/eca6f5ca-eba2-4998-80b2-6b1e6a477f70)

And  2xAAA batteries in the [BTH01](https://pvvx.github.io/BTH01/):
![image](https://github.com/user-attachments/assets/1f4641f5-095a-4515-96ad-d57691150023)

Now the market is filled with cheaper devices that work only if you constantly change batteries...
If you are satisfied with the replacement of batteries more than once every six months, you can publish the firmware. Let users suffer :)

---

An acceptable maximum average current from the CR2032 I think 15 μA. With a pulsed current of not more than 10 mA. This already reduces the maximum energy CR2032 to 100 mA/h and works in the maximum of up to 1 year.
Everything exceeding these values ​​is the trash.

Another example of trash: [ZSS-ZK-THL](https://pvvx.github.io/ZSS-ZK-THL) - E-ink Moes ZigBee "Smart Brightness Thermometer".
The chip during the transfer can consume 200 mA in accordance with the specification. At the same time, in sleep mode, the device has a current of less than 7 μA. 
That is, voltage failures from CR2032 are guaranteed. For this reason, it is impossible to ensure the work of this garbage with standard Zigbee coordinators. Constant rebooting occurs ...

**Reply by sjoram** [reply]
> the fw on such device. It took a while to map the LCD but I ha

I too have been duped into a box of these devices and have been scouring the web for days. Any help with loading something up would be great otherwise they are for the bin which would be a shame :(

**Comment by mpolak77** [comment]
unfortunately I have also bought a few of those, before I realized that "not so good" clones do exist. @gabbla I would like to give it try at least.

**Comment by gmag11** [comment]
May you share a build that may be tested on these devices?

**Comment by gabbla** [comment]
I'm out of office these days, I'll try to share a binary asap and then I'll submit a pr with changes

**Reply by gmag11** [reply]
That's not urgent at all. Thanks

**Reply by gabbla** [reply]
Edit: the project author asked to remove the binary since it's not possible to inspect what's shown on display (see the replays below) 

~~Here is the firmware: [deleted]~~ 
Note that the PHY6252 does not have an hardware reset pin, so you need to power the device at the right moment.
During my test I used the script provided in this repo as

```shell
python3 rdwr_phy62x2.py -p /dev/ttyUSB0 -e -r wh bthome_phy6222/build/bthome_phy6222.hex
```
And powered the board when the tool printed "Connecting..". It takes several tries before succeeding. 

cc @mpolak77

**Reply by mpolak77** [reply]
does this do the whole thing (what is normally done by using BOOT FW and OTA FW?)

**Reply by gabbla** [reply]
I never had time to try the whole thing, I only flashed the app over serial

**Reply by mpolak77** [reply]
seems to have worked - thx a lot!

**Reply by sjoram** [reply]
Brilliant - many thanks. I am not at home for a coupe of days so cannot test it at the moment

**Reply by pvvx** [reply]
@gabbla - Please do not publish binary firmware here without providing the source code to check what inscriptions can be displayed on the screen. The exception is the original firmware from the manufacturer.

**Reply by gabbla** [reply]
Sorry about that, I didn't know. I'll remove the binary and I'll work on a pr in the next days. Is it OK to share privately if someone want to test it?

**Reply by pvvx** [reply]
The license for the source code contained in these repositories has no restrictions, including authorship or attribution, unlike the GNU and others.
That is, you can copy and distribute even under your own name.

--

I can insert a link in the main READ.ME that support for the thermometer on PHY6252 is provided in https://github.com/gabbla or wherever you specify. It is difficult for me to support all the options alone.

---

And if you want to take part in this mess, then each device will require its own identification number ([HW ID](https://github.com/pvvx/pvvx.github.io?tab=readme-ov-file#id-numbers-of-alternative-firmware)). The ID number is especially important if a ZigBee version is released.

**Reply by gabbla** [reply]
I'd like to contribute to this project since I already have something working.  The next step on my side should be:
- Pick an hw id (34?)
- Find an appropriate device name (currently I'm using TH04 which seems to be - badly - printed on the label on the box)
- Sort out my tree and open a pr 

Does it sounds good? Anything else I should work on?

**Reply by sjoram** [reply]
> seems to have worked - thx a lot!

Hey - I managed to get a minute but struggling to get through the connection point to flash. How did you get the device to connect to accept the flash? I've tried to power up at various times but to no avail. Am I missing something? Cheers

**Reply by gabbla** [reply]
> Hey - I managed to get a minute but struggling to get through the connection point to flash. How did you get the device to connect to accept the flash? I've tried to power up at various times but to no avail. Am I missing something? Cheers

It's a pita honestly. The connection is pretty simple:
- G is your ground
- T is the chip TX, connect to your adapter RX
- R is the chip RX, connect to your adapter TX
- V is your supply

Ensure that T and R are connected properly. 
To simplify the process I used a switch between V and the 3v source, turning it on when the "Connecting..." prompt is displayed
It still requires 3/5 tries each time

```
python3 rdwr_phy62x2.py -p /dev/ttyUSB0 -e -r wh bthome_phy6222/build/bthome_phy6222.hex 
=========================================================
PHY62x2/ST17H66B Utility version 20.12.24
---------------------------------------------------------
Connecting...
PHY62x2: Release RST_N if RTS is not connected...
ST17H66B: Turn on the power...
Chip Reset Ok. Response: b'cmd>>:'
Revision: b'001340c4 6222M005'
FlashID: 1340c4, size: 512 kbytes
PHY62x2 - connected Ok
[.....]
```

**Reply by pvvx** [reply]
Since the chip has enough power from the RX/TX inputs in sleep mode, it is advisable to connect + power, and when starting the script, turn on GND. The same as with ST17H66B.

> Pick an hw id (34?)

Ok.

**Reply by sjoram** [reply]
So it worked! And is being picked up immediately in HA.

The clue was to use the GND and not Vcc though. Many thanks for that tip.

I did also set ttcl to a longer value in the py script to give me a bit more time to faff with the power.

**Reply by sjoram** [reply]
@gabbla the battery level is reporting 0.007V. Did you look at this for your devices? If as PVVX says the battery life is a bit crap it would be great if this worked! I'm happy enough now though so many thanks

**Reply by gabbla** [reply]
There is no adc pin connect to the battery vcc rail. I think it's possible to monitor the internal vcc, but I still have to look into that

**Reply by pvvx** [reply]
Many thermometers do not have a GPIO pin to +Bat. This is not necessary at all.
Any GPIO can output "1" and PullUp. This will be Vcc.
And the best GPIO is the one that is not routed from the crystal to an external contact on the chip package.
Otherwise, if you select the wrong firmware - upload via OTA to another thermometer, where this GPIO is connected to something, the program will switch the chip to deep-sleep. And the connection for OTA will be impossible. That is, when choosing GPIO, it is advisable to take into account all the options for chips and circuits of existing thermometers on PHY62x2.

The GPIO in the PHY62x2 that is not connected to anything causes increased power consumption - it receives interference and the internal circuit switches, creating additional current. On an empty GPIO, PullUp or PullDown must be enabled.

PowerProfiler is a must for debugging (!) Suddenly PullUp/PullDown or something else got mixed up. Then it will cause additional consumption from the battery. Without PowerProfiler you will notice this only after several months of using the thermometer...
Of the ready-made ones, not the best, but possible for use with BLE - Power Profiler Kit II (PPK2). There are no cheaper ready-made options.

**Reply by Taxom** [reply]
> Edit: the project author asked to remove the binary since it's not possible to inspect what's shown on display (see the replays below)
> 
> ~Here is the firmware: [deleted]~ Note that the PHY6252 does not have an hardware reset pin, so you need to power the device at the right moment. During my test I used the script provided in this repo as
> 
> ```shell
> python3 rdwr_phy62x2.py -p /dev/ttyUSB0 -e -r wh bthome_phy6222/build/bthome_phy6222.hex
> ```
> 
> And powered the board when the tool printed "Connecting..". It takes several tries before succeeding.
> 
> cc @mpolak77

can I get the hex file for my PHY6252? Thanks.

**Comment by pvvx** [comment]
>  I see around 18 uA drained from a CR2032 battery (without any link). 

Here is the data for reference.
PHY6252 is an analog of ST17H66B (the crystal is the same), without taking into account the display and sensor.
With a supply voltage of 3.0 V and a transmission period of 10 sec, the average consumption should be about 7.17 µA:
![image](https://github.com/user-attachments/assets/8b1b3c7d-77e1-4c76-8e5a-0234c50537e0)

In sleep mode, between transmissions - 3.45 µA:
![image](https://github.com/user-attachments/assets/49a7ced9-e29e-4051-8fb8-b564b80733e7)

The dependence of the average current on the transmission period is given here - https://github.com/pvvx/THB2/issues/94#issuecomment-2536521389

Compared to TLSR825x (sleep current is 1.8 µA), average consumption at 2.5 sec transmission period is equal to average current of PHY62x2 at 5 sec interval.

As a result, such intervals are selected by default, since from a cheap CR2032 it is about 8 months of work. And from a good CR2032 - a year. This is an average current of about 14 μA with a pulse of up to 10 mA. And this is taking into account the consumption of a good LCD driver and a sensor of the SHT4x type. Other sensors, LCD, E-Ink consume significantly more.

The PHY6252 has a higher current pulse during operation and transmission, and even with an average current of 14 μA, the battery life will be 20-30% shorter than with the PHY6222.

**Reply by FrankyGT** [reply]
Well I guess you were right Victor, I was sold cheap junk. Yesterday I flashed all 10 devices and also found out that 6 of them had duplicate addresses. So I can only use 4 of them... Luckily I only paid 2 euro a piece...

**Reply by pvvx** [reply]
MAC address changes in the program.
Any firmware uses 3 random bytes for MAC. The first three are usually fixed and assigned to the manufacturer.
If you erase the entire Flash, then a new MAC will be generated anew.
When copying FullFlash to another device, the MAC will match.

The probability of getting devices with the same MAC is very high.

**Comment by Taxom** [comment]
Can anyone share the firmware for this thermometer? I really do not want to throw it into garbage. 
Thanks

**Reply by Shiro-Nek0** [reply]
i got it from this comment: https://github.com/pvvx/THB2/discussions/82#discussioncomment-11677934
press "edited" and check the older version of the comment

**Reply by Taxom** [reply]
I managed to flash and connect it to HA, but the battery status doesn't work (voltage always shows 0.002V) and the smiley face doesn't appear on the screen. Everything else works. Is it the same for you?

**Reply by Shiro-Nek0** [reply]
Yes, but for now it works great

**Reply by pvvx** [reply]
Looks like @gabbla hasn't fully finished the firmware yet.


To include this version of the thermometer in support, need a map of the LCD segments, a complete schematic of the device and someone who can perform tests for each new firmware version. But so far this has not been observed.

Wait for @gabbla to make his own version.

**Reply by sjoram** [reply]
To be honest I think Victor was spot on with his comments on these devices. Gabbla did great work and was much appreciated but I did not find the devices very good. The humidity readings were up to 10% different when sitting beside each other and the temperatures were 2 deg different. I have since bought 5 BTH05s and while I know they are crap for the battery life they are within 0.5 deg and 1% humidity of each other sitting on the bench!

**Reply by mrburns42** [reply]
I have three of these units flashed with gabbla's firmware.  Thank you for that firmware.    My experiences are different.   The units track temperature within a few tenths of a degree C and they correlate to my IR thermometer.   The humidity readings are within 5%.   The spec sheet included with the device is +-1 degrees C and +-5% humidity.   Calling them junk is a little harsh.  The design engineers were given a parts count and a cost to hit.   They did a reasonable job.  (I am not the designer!)     Just look at the photos, if you are designing fakes, then you don't spend money on silk screen and you don't route a thermal island around the temperature sensor.

**Reply by pvvx** [reply]
When this thermometer works for more than a year from CR2032, then you can discuss it.

As of today, gabbla does not want to complete the firmware or transfer the previously described information about this device.
To be able to use the existing firmware, you are advised to copy the current version of PHY62x2BTHome.html and other necessary files.
When changing, due to the appearance of new versions, this firmware may not be supported in PHY62x2BTHome.html.
I won’t specifically rule out anything, but I won’t have anything to test this firmware on.

**Reply by avataru** [reply]
@gabbla if you are still reading this, please reconsider...

**Reply by gabbla** [reply]
Hey all, I want to complete the work but I don't have enough time to spare on this project. Since it's gathering some attention I'll try my best to push at least a pr so it can be tested and adjusted

**Comment by gabbla** [comment]
So I've pushed my changes at [gabbla/THB2](https://github.com/gabbla/THB2/tree/support_th04). The config is falling back to `DEVICE_TH04` if  not specified. To build and flash:

```shell
make -C bthome_phy6222
python3 rdwr_phy62x2.py -p /dev/ttyUSB0 -e -r wh bthome_phy6222/build/bthome_phy6222.hex
```

I've also fixed the battery reading thanks to Victor's (really clever) suggestion.
The firmware is based on v1.9 because I had some issues with v2.0. It is still usable tho.

**Reply by Taxom** [reply]
Thank you! Even though it took 2 hours of chatting with GPT and installing 4GB of software, I was able to build and install the firmware. It works! I think the battery consumption has increased, but it’s possible that my battery is just really old.

**Comment by climberhunt** [comment]
Worked great for me. Ordered 5 of these without knowing which one's I'd end up with.
I was using Ubuntu 24.04. 

```
apt-get install python3-pip
pip3 install requirements.txt
apt-get install gcc-arm-none-eabi
make -C bthome_phy6222
python3 rdwr_phy62x2.py -p /dev/ttyUSB0 -e -r wh bthome_phy6222/build/bthome_phy6222.hex

```
Tip of connecting GND just after hitting return on the last command above worked almost every time. 

Now to see if I can find the more efficient models. :)

**Comment by pvvx** [comment]
It is not yet possible to include the described the rmometer in support.
1. The name of the thermometer is not indicated. Usually the name is written on the box or in the enclosed instructions.
2. There is no photograph or drawing of all LCD display segments.
3. There are no photos of the PCB from both sides for the user to identify this thermometer.
4. The posted firmware does not contain a map of LCD segments. The code does not display the battery symbol, clock, etc.

(And the price of this thermometer is 1.5 times more than that of Xiaomi Mijia LYWSD03MMC.
I ordered it, I'll wait for it to arrive...)

**Comment by pvvx** [comment]
As expected, TH04 has the highest consumption of all presented in this repository.
![image](https://github.com/user-attachments/assets/46439171-8242-42f0-8af0-5f3851242009)
![image](https://github.com/user-attachments/assets/a06c985f-6808-4ce9-bf7c-bc719f01294f)

A few hours ago I got these [TH04](https://pvvx.github.io/TH04/).
I created the code and already added it to the general list.
I'll update github now.
The topic is closed.

**Comment by tymscar** [comment]
Hello there!
I was also one of the unlucky ones that got this board, intead of the other nice ones with RST pin on them.
I managed to find this discussion luckily, but I can't flash the firmware.

I have tried the trick with plugging in the ground pin just as soon as I run:
`python3 rdwr_phy62x2.py -p /dev/tty.usbserial-2120 -e -r wh bthome_phy6222/build/bthome_phy6222.hex`

But all I get back is either:
```
=========================================================
PHY62x2/ST17H66B/TG7100B Utility version 08.02.25
---------------------------------------------------------
Connecting...
PHY62x2/TG7100B: Release RST_N if RTS is not connected...
ST17H66B: Turn on the power...
Chip Reset error! Response: b''
Check connection TX->RX, RX<-TX, RTS->RESET, TM, and Chip Power!
```

Or, worse:
```
=========================================================
PHY62x2/ST17H66B/TG7100B Utility version 08.02.25
---------------------------------------------------------
Connecting...
PHY62x2/TG7100B: Release RST_N if RTS is not connected...
ST17H66B: Turn on the power...
Traceback (most recent call last):
  File "/Users/tymscar/Downloads/THB2/rdwr_phy62x2.py", line 835, in <module>
    main()
  File "/Users/tymscar/Downloads/THB2/rdwr_phy62x2.py", line 655, in main
    if not phy.Connect(args.baud):
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tymscar/Downloads/THB2/rdwr_phy62x2.py", line 261, in Connect
    self._port.setDTR(False) #TM  (hi)
    ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/rxc924pd4nvh22kzlr0dn3nr8wb2s340-python3.12-pyserial-3.5/lib/python3.12/site-packages/serial/serialutil.py", line 603, in setDTR
    self.dtr = value
    ^^^^^^^^
  File "/nix/store/rxc924pd4nvh22kzlr0dn3nr8wb2s340-python3.12-pyserial-3.5/lib/python3.12/site-packages/serial/serialutil.py", line 473, in dtr
    self._update_dtr_state()
  File "/nix/store/rxc924pd4nvh22kzlr0dn3nr8wb2s340-python3.12-pyserial-3.5/lib/python3.12/site-packages/serial/serialposix.py", line 715, in _update_dtr_state
    fcntl.ioctl(self.fd, TIOCMBIC, TIOCM_DTR_str)
OSError: [Errno 6] Device not configured
```

Now I am very suspicious of that Device not configured error, because I get the same if I just cat the serial port and reset ground.
However I can read stuff just fine from my device with minicom.

Any ideas? I have also tried both building the firmware and using `bthome_phy6222/build/bthome_phy6222.hex` as well as using the hex posted last year.

Thanks you!

Edit: I have also tried it with @gabbla repo on that branch, and it's the same.

Edit 2: Just to make sure this is not a fluke with macOS, I have tried it from my raspberry pi, and after 10 tries, the best I got was:
```
=========================================================
PHY62x2/ST17H66B/TG7100B Utility version 08.02.25
---------------------------------------------------------
Connecting...
PHY62x2/TG7100B: Release RST_N if RTS is not connected...
ST17H66B: Turn on the power...
Traceback (most recent call last):
  File "/root/THB2/rdwr_phy62x2.py", line 835, in <module>
    main()
  File "/root/THB2/rdwr_phy62x2.py", line 655, in main
    if not phy.Connect(args.baud):
  File "/root/THB2/rdwr_phy62x2.py", line 269, in Connect
    read = self._port.read(6)
  File "/usr/local/lib/python3.9/dist-packages/serial/serialposix.py", line 595, in read
    raise SerialException(
serial.serialutil.SerialException: device reports readiness to read but returned no data (device disconnected or multiple access on port?)
```

**Reply by FrankyGT** [reply]
Just an update, as it looks there is a reason why this device uses twice the amount the energy.
Let me explain. First of all, I just replaced the first number of batteries. Some of them were still sending updates, but the display turned off. I still have some running, one still reports 40% battery, so I think I used different (better?) batteries in some sensors.
So you were right about changing the batteries about every 6 months.
So I also followed your advice and bought ten additional sensors, but this time I bought 10 THB1 sensors.
Now, I have 2 sensors in the same room, 2 meters apart, and I get these readings  : 
THB1 : 
<img width="1330" height="983" alt="Schermafbeelding 2025-08-05 100518" src="https://github.com/user-attachments/assets/aefd402e-ad76-42e0-822b-d3df2251509b" />
TH05 : 
<img width="1334" height="988" alt="Schermafbeelding 2025-08-05 100602" src="https://github.com/user-attachments/assets/1089c65c-2f64-49da-8839-e975df504b71" />

When I tried to replace a TH05 sensor,which worked perfectly fine for weeks, with a THB1 sensor I got this : 

<img width="1313" height="978" alt="Schermafbeelding 2025-08-05 101143" src="https://github.com/user-attachments/assets/a530938f-758f-4c56-85ec-66cf14409181" />

So I actually prefer the TH05 sensors over the THB1

FYI : You don't see these missing measurements when using the home assistant viewer, as this will just connect the dots. Using grafana you can set this to your liking

Also, battery life of THB1 doesn't look like it will hold up a year either : 

<img width="1316" height="676" alt="Schermafbeelding 2025-08-05 103956" src="https://github.com/user-attachments/assets/88276206-ccce-418a-a3d1-82ffa19b3fab" />

**Reply by tymscar** [reply]
I think you replied to the wrong comment :)

**Reply by FrankyGT** [reply]
Indeed, I see now... Well I assume victor will read it anyway... :-)

**Reply by FrankyGT** [reply]
And to help you with your problem, I used hotglue to glue pins together to fit the points on the board. then I connected it to a breadboard with a single button for the ground pin. I also struggled in the beginning, but once you find the right timing, I  could do the rest almost all in one go...

**Reply by tymscar** [reply]
Well, as far as I can tell the connection is pretty good on mine. Any tips on the timing?
